pkg/upstream/cluster/loadbalancer.go

```diff
@@ -779,10 +788,22 @@ func (lb *peakEwmaLoadBalancer) unweightedPeakEwmaScore(h types.Host) float64 {
 	stats := h.HostStats()
 
 	duration := stats.UpstreamRequestDurationEWMA.Rate()
-	// None of the active requests returned, or the metrics is disabled
+
+	// None of the active requests returned, try to use cluster duration as default.
 	if duration == 0 {
-		duration = float64(lb.defaultDuration)
+		duration = h.ClusterInfo().Stats().UpstreamRequestDurationEWMA.Rate()
+		// None of the active requests returned in cluster, or the metrics is disabled
+		if duration == 0 {
+			duration = float64(lb.defaultDuration)
+		}
+	}
+
+	biasedActiveRequest := float64(stats.UpstreamRequestActive.Count()) + 1
+	if biasedActiveRequest == 1.0 || lb.activeRequestBias == 0.0 {
+		biasedActiveRequest = 1
```

Comment:
Why set biasedActiveRequest to 1 when `if biasedActiveRequest == 1.0`?

Reply:
avoid too many branches

Reply:
I take a look at Go's Pow implementation:
https://github.com/golang/go/blob/e335a2665f5e322a7da8baa22fe816b6ef9aaf24/src/math/pow.go#L57-L60
There are already additional special handling for these cases.

Do the additional branches in the caller's code improve the performance?

Reply:
Thank you very much, I haven't paid attention to the implementation of Pow before, here is a redundant optimization